src/bpftrace.cpp

```diff
@@ -120,29 +128,79 @@ int BPFtrace::add_probe(ast::Probe &p)
     }
     else
     {
-      attach_funcs.push_back(attach_point->func);
+      if (probetype(attach_point->provider) == ProbeType::usdt && !attach_point->ns.empty())
+        attach_funcs.push_back(attach_point->ns + ":" + attach_point->func);
+      else
+        attach_funcs.push_back(attach_point->func);
     }
 
-    for (auto func : attach_funcs)
+    for (auto func_ : attach_funcs)
     {
+      std::string full_func_id = func_;
+      std::string func_id = func_;
+
+      // USDT probes must specify both a provider and a function name for full id
+      // So we will extract out the provider namespace to get just the function name
+      if (probetype(attach_point->provider) == ProbeType::usdt )
+      {
+        std::string ns = func_id.substr(0, func_id.find(":"));
+        func_id.erase(0, func_id.find(":")+1);
+        // Set attach_point ns to be a resolved namespace in case of wildcard
+        attach_point->ns = ns;
+        // Set the function name to be a resolved function id in case of wildcard
+        attach_point->func = func_id;
+      }
+
       Probe probe;
       probe.path = attach_point->target;
-      probe.attach_point = func;
+      probe.attach_point = func_id;
       probe.type = probetype(attach_point->provider);
       probe.orig_name = p.name();
       probe.ns = attach_point->ns;
-      probe.name = attach_point->name(func);
+      probe.name = attach_point->name(func_id);
       probe.freq = attach_point->freq;
       probe.loc = 0;
-      probe.index = attach_point->index(func) > 0 ?
-          attach_point->index(func) : p.index();
+      probe.index = attach_point->index(full_func_id) > 0 ?
+          attach_point->index(full_func_id) : p.index();
       probes_.push_back(probe);
     }
   }
 
   return 0;
 }
 
+// FIXME should this really be a separate function?
```

Comment:
I rcould just refactor the existing find_wildcard_matches to accept more arguements, but I want to be able to glob on provider as well so I copy-pasta'd the function to add support as to not risk affecting behavior for non USDT probes.

Reply:
I think the existing find_wildcard_matches should glob on `prefix` as well (tracepoints would also benefit from this), and the USDT provider should be passed as prefix.

Reply:
I can follow up with this change once we merge this PR though, since it's kinda related to #557